src/lib.rs

```diff
@@ -417,7 +418,7 @@ impl CoreBPE {
         special_tokens_encoder: HashMap<String, Rank>,
         pattern: &str,
     ) -> PyResult<Self> {
-        let regex = Regex::new(pattern)
+        let regex = RegexBuilder::new(pattern).backtrack_limit(100_000).build()
```

Comment:
doesn't work for values bigger than a million - see https://github.com/fancy-regex/fancy-regex/issues/134

I've set it lower for now, hoping we'll be able to fix the whitespace problem